src/phospho.ts

```diff
@@ -5,24 +5,83 @@ import { PhosphoInit, LogContent, LogEvent, UserFeedback } from "./types";
 import { getInputOutput, extractMetadataFromInputOutput } from "./extractor";
 import { BASE_URL } from "./config";
 import { sendUserFeedback } from "./user-feedback";
+import { hashCode } from "./utils";
 
+/**
+ * Phospho class for logging and processing events
+ * @class
+ */
 class Phospho {
+  /**
+   * @property {string} apiKey - The API key for authentication, found in your Phospho dashboard
+   */
   apiKey: string;
+  /**
+   * @property {string} projectId - The project id for the project you want to log events to, found in your Phospho dashboard
+   */
   projectId: string;
+  /**
+   * @property {number} tick - The delay in milliseconds for debounced operations
+   * @default 500
+   */
   tick: number = 500;
+  /**
+   * @property {string} baseUrl - The base URL for API requests, defaults to the Phospho API
+   * @default "https://api.phospho.ai/v2"
+   */
   baseUrl: string = BASE_URL;
-  // context: any;
+  /**
+   * @property {string} path_to_hash - The path to your codebase to hash for versioning
+   * @example "process.cwd()"
+   */
+  path_to_hash: string | null = null;
 
   // Queue of log events as a Mapping of {taskId: logEvent}
+  /**
+   * @property {Map<string, LogEvent>} logQueue - The queue of log events to be sent to Phospho
+   */
   logQueue = new Map<string, LogEvent>();
+  /**
+   * @property {string | null} latestTaskId - The latest task id generated
+   */
   latestTaskId: string | null = null;
+  /**
+   * @property {string | null} latestSessionId - The latest session id generated
+   */
   latestSessionId: string | null = null;
 
-  // constructor(context?) {
-  //   this.context = context;
-  // }
+  // Version ID
+  /**
+   * @property {Promise<string> | string | null} version_id - The version id of your app, to be used for versioning
+   */
+  version_id: Promise<string> | string | null = null;
```

Comment:
can you please use camelCase instead

Reply:
Here you ask for it to be string, but the implementation actually support string[] list. Can you normalize the types ?